pkg/apis/core/v1alpha1/types_schedulingprofile.go

```diff
@@ -46,15 +45,74 @@ type SchedulingProfileList struct {
 }
 
 type SchedulingProfileSpec struct {
-	// Plugins is the list of plugins to use when scheduling a resource
-	Plugins []SchedulingPlugin `json:"plugins"`
+	// Plugins specify the set of plugins that should be enabled or disabled.
+	// Enabled plugins are the ones that should be enabled in addition to the
+	// default plugins. Disabled plugins are any of the default plugins that
+	// should be disabled.
+	// When no enabled or disabled plugin is specified for an extension point,
+	// default plugins for that extension point will be used if there is any.
+	Plugins *Plugins `json:"plugins"`
+	// PluginConfig is an optional set of custom plugin arguments for each plugin.
+	// Omitting config args for a plugin is equivalent to using the default config
+	// for that plugin.
+	// +optional
+	PluginConfig []PluginConfig `json:"pluginConfig,omitempty"`
 }
 
-type SchedulingPlugin struct {
-	// Name is the name of the scheduler plugin
-	Name string `json:"name"`
+// Plugins include multiple extension points. When specified, the list of plugins for
+// a particular extension point are the only ones enabled. If an extension point is
+// omitted from the config, then the default set of plugins is used for that extension point.
+type Plugins struct {
+	// Filter is the list of plugins that should be invoked during the filter phase.
+	// +optional
+	Filter PluginSet `json:"filter,omitempty"`
+	// Score is the list of plugins that should be invoked during the score phase.
+	// +optional
+	Score  PluginSet `json:"score,omitempty"`
+	// Select is the list of plugins that should be invoked during the select phase.
+	// +optional
+	Select PluginSet `json:"select,omitempty"`
+}
+
+// PluginSet contains the list of enabled and disabled plugins.
+type PluginSet struct {
+	// Enabled specifies plugins that should be enabled in addition to the default plugins.
+	// Enabled plugins are called in the order specified here, after default plugins. If they need to
+	// be invoked before default plugins, default plugins must be disabled and re-enabled here in desired order.
+	// +optional
+	Enabled []Plugin `json:"enabled,omitempty"`
+	// Disabled specifies default plugins that should be disabled.
+	// +optional
+	Disabled []Plugin `json:"disabled,omitempty"`
+}
+
+// Plugin specifies a plugin type, name and its weight when applicable. Weight is used only for Score plugins. Type
+// should be ommited when referencing in-tree plugins.
+type Plugin struct {
+	// Type defines the type of the plugin.
+	// +kubebuilder:validation:Enum=Webhook
```

Comment:
Is it better to specify this on the type instead?

Reply:
fixed

Reply:
did you push?

Reply:
yes

Reply:
I still see the marker comment on the field 
![image](https://user-images.githubusercontent.com/44059321/232226108-d1ac3983-1a2b-48d3-8a3d-5424e5e6809a.png)
and this comment is not marked as outdated

Reply:
fixed now